content/_guides/channelmodes.md

```diff
@@ -4,25 +4,224 @@ category: chanop
+
+```irc
+/msg ChanServ SET <channel> MLOCK <mode>
+```
+
+For example, to ensure setting the topic on `#foo` is always restricted to
+operators:
+
+```irc
+/msg ChanServ SET #foo MLOCK +t
+```
+
+With this set, if someone were to attempt `/mode #foo -t`, `ChanServ` would
+immediately revert the change with `/mode #foo +t`.
+
+To list modes currently enforced by `ChanServ`:
+
+```irc
+/msg ChanServ INFO <channel>
+```
+
+### Available channel modes
+
+Channel operators can set the following modes.
+<!-- markdownlint-disable MD013 -->
 {% assign channel_modes = site.data.channelmodes | where_exp:"item","item.restricted != true" %}
+<!-- markdownlint-enable MD013 -->
 
 {%- include modetable.md modes=channel_modes -%}
 
-## Restricted channel modes
-
-The following channel modes can only be added by Libera.Chat staff.
+### Restricted channel modes
 
+Only Libera.Chat staff can set the following modes.
+<!-- markdownlint-disable MD013 -->
 {% assign restricted_modes = site.data.channelmodes | where: 'restricted', true %}
+<!-- markdownlint-enable MD013 -->
 
 {%- include modetable.md modes=restricted_modes -%}
+
+## ChanServ user flags
+
+`ChanServ` user flags grant privileges for `ChanServ` functions. They apply to
+users identified by nick or hostmask. 
+
+User flags are similar to modes, but they control what users can do via
+`ChanServ` rather than what they can do directly with the channel. Because
+`ChanServ` is always opped, it can perform actions on behalf of users who are
+not opped, according to privileges specified by their user flags.
+
+ChanServ user flags control:
+
+- Who can use `ChanServ` to moderate users (`v`, `o`, `b`, `r`, `e`, `f`)
+- Who is voiced or opped automatically by `ChanServ` (`V`, `O`)
+- Who can use `ChanServ` commands to manage the channel (`s`, `i`, `R`, `t`,
+  `A`, `S`, `F`)
+
+### Requirements
+
+- Your channel must be registered.
+- `ChanServ` must have joined your channel. See [preserving channel
+  modes](#preserving-channel-modes).
+- To set `ChanServ` user flags for users other than yourself, you must be
+  flagged a founder (`+F`). 
+
+### Command summary
+<!-- markdownlint-disable MD013 -->
+| Purpose          | Command                                                  |
+|------------------|----------------------------------------------------------|
+| List user flags  | `/msg ChanServ FLAGS <channel> <nick\|hostmask>`         |
+| Set user flags   | `/msg ChanServ FLAGS <channel> <nick\|hostmask> +<flags>`|
+| Unset user flags | `/msg ChanServ FLAGS <channel> <nick\|hostmask> -<flags>`|
+<!-- markdownlint-enable MD013 -->
+
+### Granting limited user privileges
+
+User flags are useful for allowing users to take actions through `ChanServ`
+without granting them full privileges on the channel itself. For example, you
+might restrict topic changes on `#foo` to ops, but then allow non-op user
+`bar` to ask `ChanServ` to set the topic: 
+
+```irc
+/mode #foo +t
+/msg ChanServ FLAGS #foo bar*!*@user/bar +t
```

Comment:
It's better to set flags on an account, not a hostmask, since hostmasks can change

Reply:
Indeed. Thank you!